### ISSUE TITLE
Bug: listContainersCmd() makes docker client block on every further request

### ISSUE THREAD

**Issue by Copperazide** [docker-java/docker-java#1748]
I'm not entirely sure if it's just me doing something wrong or if I found a bug.

I'm calling the listContainersCmd() method on a DockerClient object. When some time passes before I make another request on the same docker client object, the docker client always blocks, no matter what request it is.

To reproduce:

    dockerClient.listContainersCmd().exec();

    try {
        Thread.sleep(5000); //let some time pass
    } catch (InterruptedException e) {
        e.printStackTrace();
    }

    dockerClient.versionCmd().exec(); //this line will block

When almost no time passes after the listContainersCmd() call, it works.

Is there a solution to this? Right now the only thing I can do is create a new docker client since the old one seems bugged.

**Comment by bram-pkg** [comment]
I have this same issue.

**Comment by bram-pkg** [comment]
@Copperazide switching to the ZeroDep transport fixed it for me.

<https://mvnrepository.com/artifact/com.github.docker-java/docker-java-transport-zerodep>

It seems that <https://mvnrepository.com/artifact/com.github.docker-java/docker-java-transport-httpclient5> is buggy, somehow.

**Comment by zotornit** [comment]
Same problem. (Zerodep fixes it)

I tried with `3.2.11` and same problem. Maybe it is an API 1.41 issue?

I run docker-java within `openjdk:17-alpine` based container

```
docker version
Client: Docker Engine - Community
 Version:           20.10.12
 API version:       1.41
 Go version:        go1.16.12
 Git commit:        e91ed57
 Built:             Mon Dec 13 11:45:37 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.12
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.12
  Git commit:       459d0df
  Built:            Mon Dec 13 11:43:46 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.12
  GitCommit:        7b11cfaabd73bb80907dd23182b9347b4245eb5d
 runc:
  Version:          1.0.2
  GitCommit:        v1.0.2-0-g52b36a2
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

```

**Comment by tongxin-77** [comment]
> @Copperazide switching to the ZeroDep transport fixed it for me.
> 
> https://mvnrepository.com/artifact/com.github.docker-java/docker-java-transport-zerodep
> 
> It seems that https://mvnrepository.com/artifact/com.github.docker-java/docker-java-transport-httpclient5 is buggy, somehow.

it works，thanks.

**Comment by bsideup** [comment]
zerodep is literally a shaded version of httpclient5. Perhaps your dependencies override the version of Apache HttpClient5 and somehow break it.

**Comment by dimovelev** [comment]
Seems to be what i experience in #1789

**Comment by bsideup** [comment]
Should be fixed in 3.2.13
